### ISSUE TITLE
Extend `prefer-explicit-assert` to report `findBy*` and `waitFor`

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#409]
With `prefer-explicit-assert` enabled, `getBy*` usages without `expect` are reported, but `findBy*` and `waitFor` are not reported:

```
    screen.getByText('hey') // reported
    await waitFor(() => screen.getByText('hey')) // not reported
    await screen.findByText('hey') // not reported
```

What if we extended `prefer-explicit-assert` to report all of the above cases? 🤔

**Comment by Belco90** [comment]
Hey @zaicevas! Few comments about this improvement idea.

What about `findBy*` usage saving returned element within a var? That should not be reported.

Not really sure about reporting `waitFor` + `getBy*` query, since inside `waitFor` callback the user might have other necessary stuff there.

I'd go just for reporting `findBy*` usages for now. What do you think?

**Comment by zaicevas** [comment]
> What about findBy* usage saving returned element within a var? That should not be reported.

Makes sense to me, since `const foo = screen.getByText('hey')` is not reported and there's no reason the behaviour should be different for `findBy*`

> Not really sure about reporting waitFor + getBy* query, since inside waitFor callback the user might have other necessary stuff there.

Can you give an example to better understand what you have in mind?

> I'd go just for reporting findBy* usages for now. What do you think?

Sure, I'd imagine these 2 cases having separate PRs anyway.

**Comment by Belco90** [comment]
> > Not really sure about reporting waitFor + getBy* query, since inside waitFor callback the user might have other necessary stuff there.
> 
> Can you give an example to better understand what you have in mind?

I don't have a really clear example in mind, to be honest. This is similar to `no-unnecessary-act` rule assuming there could be non-react-testing-library things inside `act`, in which case we need to allow.

This is a stupid example, but I'm not sure it should be reported:

```js
const promiseA = new Promise()
const promiseB = new Promise()
// ...
await waitFor(async () => {
  await promiseA;
  await promiseB;
  screen.getByRole('button')
})
```

It's obvious it should be rewritten to just do `await screen.findByRole('button')` instead of the whole `waitFor`, but could be a legit case were using `waitFor` +  awaiting promises +  asserting with implicit `getBy*`? I think there must be for sure, so I'd say just leave the rule complaining about that standalone `getBy*` query should be enough.

What do you think?
